xonsh/xontribs_meta.py

```diff
@@ -0,0 +1,578 @@
+"""
+This modules is the place where one would define the xontribs.
+"""
+
+import functools
+import typing as tp
+
+
+class _XontribPkg(tp.NamedTuple):
+    """Class to define package information of a xontrib.
+
+    Attributes
+    ----------
+    install:
+        a mapping of tools with respective install commands. e.g. {"pip": "pip install xontrib"}
+    license:
+        license type of the xontrib package
+    name:
+        full name of the package. e.g. "xontrib-argcomplete"
+    url:
+        URL to the homepage of the xontrib package.
+    """
+
+    install: tp.Dict[str, str]
+    license: str = ""
+    name: str = ""
+    url: tp.Optional[str] = None
+
+
+class Xontrib(tp.NamedTuple):
+    """Meta class that is used to describe xontribs.
+
+    Attributes
+    ----------
+    url: str
+        url
+    description: str
+        short description about the xontrib.
+    package: _XontribPkg
+        pkg information for installing the xontrib
+    tags: tuple of str
```

Comment:
These need a space after the name, as per numpydoc, `tags : tuple of str`

Reply:
removed the types. The attributes are already annotated.